docs/advanced/instrumentation.md

```diff
@@ -121,35 +121,177 @@ If we wouldn't do this, the result for an async data fetcher would always be 0,
+
+
+### Error Counter: gql.error
+
+Captures the number of GraphQL errors encountered during query execution of a query or mutation.
+Remember that one _graphql_ request can have multiple errors.
+
+**Name:** `gql.error`
+
+**Tags:**
+
+| tag name          | description |
+| ----------------- | ----------- |
+| `gql.errorCode`   | The GraphQL error code, such as `VALIDATION`, `INTERNAL`, etc.|
+| `gql.path`        | The sanitized query path that resulted in the error.     |
+| `gql.errorDetail` | Optional flag containing additional details, if present. |
+
+### Data Loader Timer: gql.dataLoader
+
+Captures the elapsed time for a data loader invocation for a batch of queries.
+This is useful if you want to find data loaders that might be responsible for poor query performance.
+
+**Name:** `gql.dataLoader`
+
+**Tags:**
+
+| tag name              | description |
+| --------------------- | ----------- |
+| `gql.loaderName`      | The name of the data loader, may or may not be the same as the type of entity. |
+| `gql.loaderBatchSize` | The number of queries executed in the batch. |
+
+
+### Data Fetcher Timer: gql.resolver
+
+Captures the elapsed time of each data fetcher invocation.
+This is useful if you want to find data fetchers that might be responsible for poor query performance.
+
+!!! note
+    This metric is not available if the data is resolved via a Batch Loader.
+
+**Name:** `gql.resolver`
+
+**Tags:**
+
+| tag name    | description |
+| ----------- | ----------- |
+| `gql.field` | Name of the data fetcher. This has the `${parentType}.${field}` format as specified in the `@DgsData` annotation. |
+
+
+#### Data Fetcher Timer as a Counter
+
+The data fetcher, or resolver, timer can be used as a counter as well.
+If used in this manner it will reflect the number of invocations of each data fetcher.
+This is useful if you want to find out which data fetchers are used often.
+
+### Further Tag Customization
+
+You can customize the tags applied to the metrics above by providing *beans* that implement the following functional interfaces.
+
+| Interface                    | Description |
+| ---------------------------- | ----------- |
+| `DgsContextualTagCustomizer` | Used to add common, contextual tags. Example of these could be used to describe the deployment environment, application profile, application version, etc |
+| `DgsExecutionTagCustomizer`  | Used to add tags specific to the [ExecutionResult] of the query. The [SimpleGqlOutcomeTagCustomizer] is an example of this.|
+| `DgsFieldFetchTagCustomizer` | Used to add tags specific to the execution of _data fetchers_. The [SimpleGqlOutcomeTagCustomizer] is an example of this as well.|             |
+
+
+### Configuration
+
+#### management.metrics.dgs-graphql.enabled
+
+Enables the metrics provided out of the box.
+**Defaults** to `true`.
+
+#### management.metrics.dgs-graphql.tag-customizers.outcome.enabled
+
```

Comment:
Maybe add a sentence or two clarifying that  for GraphQL, the success or failure is based on whether there were any graphql errors, even though the Http request by itself is succesful.